src/state_space/mod.rs

```diff
@@ -225,6 +225,16 @@ where
             }
         }
 
+        for variant in amm_variants.keys() {
+            while let Some(amms) = amm_variants.get(variant) {
+                for amm in amms.iter() {
+                    for event in amm.sync_events() {
+                        filter_set.insert(event);
+                    }
+                }
+            }
+        }
```

Comment:
`amm_variants` will always be empty here since the logic above drains the HashMap.

```rust
     // Sync remaining AMM variants
        for (_, remaining_amms) in amm_variants.drain() {
            for mut amm in remaining_amms {
                let address = amm.address();
                amm = amm.init(chain_tip, self.provider.clone()).await?;
                state_space.state.insert(address, amm);
            }
        }
```

Instead you can use `self.amms` which will add all sync events for individual AMMs provided to the `StateSpaceBuilder`.

```rust
for amm in self.amms.iter() {
    for event in amm.sync_events() {
        filter_set.insert(event);
    }
}
```